scripts/iso_build.sh

```diff
@@ -58,464 +15,600 @@ OLD_ISO_TAG=""  # used to remove OLD ISO images the local dir
+        || exit 1
+
+      safe_run 10 ssh "${server}" \
+        "${ssh_dir}"/iso_html_generator/gen_html/gen.sh \
+        || exit 1
+
+      ) 9> /tmp/.iso_build.sh.move_to_mirrors.lock || return 1
+      return 0
+    fi
 }
 
 docker_clean() {
 
-	# Best effort - cleaning orphaned containers
-	docker ps -a -q | xargs -n 1 -I {} sudo docker rm {}
+  # Best effort - cleaning orphaned containers
+  docker ps -a -q | xargs -n 1 -I {} sudo docker rm {}
+
+  # Best effort - cleaning orphaned images
+  local images=$(docker images | grep '<none>' | tr -s ' ' | cut -d ' ' -f 3)
+  if [ -n "${images}" ]; then
+    docker rmi ${images}
+  fi
 
-	# Best effort - cleaning orphaned images
-	local images=$(docker images | grep '<none>' | tr -s ' ' | cut -d ' ' -f 3)
-	if [ -n "${images}" ]; then
-		docker rmi ${images}
-	fi
+}
 
+update_docker_companion() {
+
+   local HOST_ARCH="amd64"
+
+   if [[ ! `which docker-companion 2>/dev/null` ]] ; then
+     [ `which ./docker-companion 2>/dev/null` ] || {
+       echo >&2 "Fetching docker-companion for you..."
+       curl -s https://api.github.com/repos/mudler/docker-companion/releases/latest \
+         | grep "browser_download_url.*${HOST_ARCH}" \
+         | cut -d : -f 2,3 \
+         | tr -d \" \
+         | wget -i - -N -O docker-companion
+       chmod +x docker-companion
+     }
+   fi
 }
 
-build_spinbase() {
-	# Generating the spinbase with docker
-
-	local docker_image=${1-sabayon/spinbase-amd64:latest}
-	local undocker_output_directory=${2-sources/amd64-docker-spinbase}
-	
-	echo "Checking if Docker is available, otherwise restarting it"
-	
-	docker images | grep -q "404 page not found" && systemctl restart docker
-	
-	echo "Building Spinbase with Docker image: "${docker_image}
-
-	# Cleaning previous generation
-	if [ -z "${undocker_output_directory}" ] || [ -z "${SABAYON_MOLECULE_HOME}" ]; then
-		echo "SABAYON_MOLECULE_HOME or undocker_output_directory not set, this is bad"
-		return 1
-	fi
-	rm -rf "${SABAYON_MOLECULE_HOME}"/"${undocker_output_directory}"
-
-	echo "Exporting the Docker image in: " ${SABAYON_MOLECULE_HOME}/${undocker_output_directory}
-
-	# Unpack the image
-	docker-companion --pull unpack --squash "${docker_image}" "${SABAYON_MOLECULE_HOME}"/"${undocker_output_directory}" || return 1
-	if [ ! -e "${SABAYON_MOLECULE_HOME}/${undocker_output_directory}/dev/urandom" ]; then
-		echo "/dev/urandom not present on unpacked chroot. creating it "
-		mknod -m 444 "${SABAYON_MOLECULE_HOME}"/"${undocker_output_directory}"/dev/urandom c 1 9 || return 1
-	fi
-	
-	docker_clean
+export_docker_rootfs () {
+
+  local docker_image=${1-${SABAYON_DOCKER_SRC_IMAGE}}
+  local undocker_output_directory=${2-${SABAYON_UNDOCKER_OUTPUTDIR}}
+  local opts=""
+
+  if [[ -z "${PULL_SKIP}"  ||  "${PULL_SKIP}" == "0" ]] ; then
+    opts="--pull"
+  fi
+
+  echo "Checking if Docker is available, otherwise restarting it"
+  docker images | grep -q "404 page not found" && systemctl restart docker
```

Comment:
This would go better i think: `systemctl show --property ActiveState docker | grep -q inactive && systemctl start docker`

Reply:
Done